components/list/demo/list.html

```diff
@@ -155,22 +156,63 @@ <h2>List (with all the stuff)</h2>
+							{ key: '4', title: 'Pedagogy Case Studies', imgSrc: 'https://s.brightspace.com/course-images/images/e5fd575a-bc14-4a80-89e1-46f349a76178/tile-high-density-max-size.jpg', supportingInfo: 'Belay yo-ho-ho keelhaul squiffy black spot yardarm spyglass sheet transom heave to. Deadlights jack lad schooner scallywag dance the hempen jig carouser broadside cable strike colors.' },
+							{ key: '5', title: 'Introduction to Painting', imgSrc: 'https://s.brightspace.com/course-images/images/63b162ab-b582-4bf9-8c1d-1dad04714121/tile-high-density-max-size.jpg', supportingInfo: 'Fluke jib scourge of the seven seas boatswain schooner gaff booty Jack Tar transom spirits. Bring a spring upon her cable holystone blow the man down spanker.' },
+							{ key: '4', title: 'Analytical Chemistry', imgSrc: 'https://s.brightspace.com/course-images/images/85f865dd-4c73-49c5-8809-c872256f7715/tile-high-density-max-size.jpg', supportingInfo: 'Trysail Sail ho Corsair red ensign hulk smartly boom jib rum gangway. Fluke jib scourge of the seven seas boatswain schooner gaff booty Jack Tar transom spirits.' },
+							{ key: '5', title: 'Plant Biology', imgSrc: 'https://s.brightspace.com/course-images/images/61f07ae4-4686-4019-8146-1604cb8ae9bf/tile-high-density-max-size.jpg', supportingInfo: 'Deadlights jack lad schooner scallywag dance the hempen jig carouser broadside cable strike colors. Bring a spring upon her cable holystone blow the man down spanker.' }
+						];
+
+						const list = document.querySelector('#allFeatures');
+
+						const createItem = (key, title, imgSrc, supportingInfo) => {
 							const item = document.createElement('d2l-list-item');
 							item.selectable = true;
-							item.key = `${++newItemCount}`;
-							item.label = 'New Item';
+							item.key = key;
+							item.label = title;
+							item.href = 'http://www.d2l.com';
 							item.innerHTML = `
-								<img slot="illustration" src="https://s.brightspace.com/course-images/images/38e839b1-37fa-470c-8830-b189ce4ae134/tile-high-density-max-size.jpg"></img>
+								<img slot="illustration" src="${imgSrc}"></img>
 								<d2l-list-item-content>
-									<div>New Item ${newItemCount}</div>
-									<div slot="supporting-info">Supporting Info</div>
+									<div>${title}</div>
+									<div slot="supporting-info">${supportingInfo}</div>
 								</d2l-list-item-content>
 							`;
-							document.querySelector('#allFeatures').appendChild(item);
+							return item;
+						};
+
+						let newItemCount = 0;
+						document.querySelector('d2l-selection-action[text="Add"]').addEventListener('d2l-selection-action-click', () => {
+							list.appendChild(createItem(`${++newItemCount}`, `New Item ${newItemCount}`, 'https://s.brightspace.com/course-images/images/38e839b1-37fa-470c-8830-b189ce4ae134/tile-high-density-max-size.jpg', 'Supporting Info'));
+						});
+
+						let lastLoadedIndex = 2;
+						list.querySelector('d2l-pager-load-more').addEventListener('d2l-pager-load-more', e => {
```

Comment:
The code for a component handling load-more would obviously be different, but this illustrates what they would be doing: loading more items, updating the state of `d2l-pager-load-more`, and calling `complete()`.

Reply:
I'm not sure if it might be easier/more typical to do this from a Lit demo component? Then you'd be "fetching" (with your timeout), adding those new items to a local state collection, which will trigger a re-render.

What's unclear to me is how calling `e.detail.complete()` would fit into that kind of workflow... I'm worried that it'll be awkward for the consumer to hold onto the promise and trigger it at the right time.

Reply:
Ok, I can set up a demo component that would be akin to what they would do.  I was sorta hoping to keep the demo component as is, since the snippet shows so many other things related to the component.  I guess it doesn't really matter too much if we don't show the snippet here, since devs will be looking at the Daylight site for examples.  Having said that, we'll have the same challenge there... showing appropriate example code without burying important aspects in the demo component.

Reply:
I've set up a demo component to show how this would work for our typical component consumers.  Obviously the data is mocked up client-side. I suspect we will want to further build this out to include some filtering, since some folks are looking for guidance on how to persist selection when filters change.